### PR TITLE
Component version added to Reports page

### PR DESCRIPTION
**Pull request by Riju19**
Displays the version of the crashing package on the respective report's page.

### CODE REVIEW

**Comment by trapas** [comment]
Can one of the admins verify this patch?

File: src/pyfaf/queries.py [hunk 1]
```diff
@@ -1287,6 +1288,17 @@ def get_crashed_unknown_package_nevr_for_report(db, report_id):
             .filter(st.ReportUnknownPackage.type == "CRASHED")
             .all())
 
+def get_crashed_package_version_for_report(db, report_id):
+    """
+    Return version for unknown packages that CRASHED in a given
+    report.
+    """
+    return (db.session.query(st.ReportUnknownPackage.version)
```
Comment (marusak) [inline src/pyfaf/queries.py]: ReportUnknownPackage is not enough (I know it works on your system well, but in general will not work) You should select from ReportPackage and from ReportUnknownPackage

File: src/pyfaf/queries.py [hunk 2]
```diff
@@ -1287,6 +1288,17 @@ def get_crashed_unknown_package_nevr_for_report(db, report_id):
...
+            .filter(st.ReportUnknownPackage.report_id == report_id)
+            .filter(st.ReportUnknownPackage.type == "CRASHED")
+            .order_by(desc('version'))
```
Comment (marusak) [inline src/pyfaf/queries.py]: This ordering is not enough either - with packages there is system epoch:version-release and all of these needs to be compared in this order

File: src/webfaf/reports.py [hunk 3]
```diff
@@ -190,6 +191,10 @@ def get_reports(filter_form, pagination):
     return r
 
 
+def get_package_version(report_id):
```
Comment (marusak) [inline src/webfaf/reports.py]: Isn't this function pointless? It is used only once and that line can be easily replaced by:
`forward['version'] = get_crashed_package_version_for_report(db,report_id)[0]`

File: src/webfaf/templates/reports/item.html [hunk 4]
```diff
@@ -64,6 +64,10 @@ <h3>Info</h3>
             {% if report.tainted %}
               <span class="label label-warning">Tainted</span>
             {% endif %}
+          </dd>
+          <dt> Component version </dt>
```
Comment (marusak) [inline src/webfaf/templates/reports/item.html]: It would be nice, if it would be clear it is the last affected version, so maybe reword it to `Last affected version`.

**Comment by Riju19** [comment]
@marusak  made the changes. Please review.

**Comment by mkutlak** [comment]
I don't think this is a right solution.

The `Last affected version` for a report which is caused by a bug in package **xyz-1.2-3** that is later fixed in **xyz-1.2-4** will show **1.2-4**.

Same problem with crashes in older Fedoras (ie. from **F24**), that were fixed. The `Last affected version` will show the latest package version (right now from **F30**).

___

I think you should try to get the latest affected version from `package_counts`.
https://github.com/abrt/faf/blob/781b4ccfb8589f0c2e8ebf9c880c308c76e87fba/src/webfaf/reports.py#L506

You will have to in some cases connect components with package names.

(Like in this case **abrt** with **abrt-plugin-bodhi** - https://retrace.fedoraproject.org/faf/reports/2434756/)

**Comment by mkutlak** [comment]
@trapas retest this please

**Comment by mkutlak** [comment]
`DEBUG: pyfaf/queries.py:1325:0: C0303: Trailing whitespace (trailing-whitespace)`

Can you fix the trailing whitespace on line 1325?

File: src/pyfaf/queries.py [hunk 5]
```diff
@@ -1317,6 +1318,26 @@ def get_crashed_unknown_package_nevr_for_report(db, report_id):
             .all())
 
 
+def get_last_affected_version(db, report_id):
+    """
+    Return the last version of a CRASHED package in a report with a given report_id
+    """
+
+    (epoch, version, release) \
+    = (db.session.query(st.Build.epoch,
```
Comment (mkutlak) [inline src/pyfaf/queries.py]: I tried to run the code without doing `faf reposync` and got this error. 
```
ERROR:flask.app:Exception on /reports/5/ [GET]
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/python3.7/site-packages/webfaf/reports.py", line 576, in item
    forward['version'] = get_last_affected_version(db, report_id)
  File "/usr/lib/python3.7/site-packages/pyfaf/queries.py", line 1335, in get_last_affected_version
    .order_by(desc('epoch'), desc('version'), desc('release'))
TypeError: cannot unpack non-iterable NoneType object
```


What if the problem comes from unknown package? (`ReportUnknownPackage`)

**Comment by mkutlak** [comment]
You don't need to create new queries. 

Just add a select for package `type` in `load_packages`.
https://github.com/abrt/faf/blob/e08cff14404d2e0e6c2cdb0591602e3840324564/src/webfaf/reports.py#L265

https://github.com/abrt/faf/blob/e08cff14404d2e0e6c2cdb0591602e3840324564/src/webfaf/reports.py#L279

And process it (filter and save `CRASHED` packages -> order/sort them -> output latest version)

https://github.com/abrt/faf/blob/e08cff14404d2e0e6c2cdb0591602e3840324564/src/webfaf/reports.py#L502

**Comment by michalfabik** [comment]
@trapas retest this please

**Comment by mkutlak** [comment]
:+1:
